src/types.ts

```diff
@@ -0,0 +1,53 @@
+export type TDay =
+  | 'Monday'
+  | 'Tuesday'
+  | 'Wednesday'
+  | 'Thursday'
+  | 'Friday'
+  | 'Saturday'
+  | 'Sunday';
+
+export interface TResource {
+  address1: string;
+  address2: string;
+  approved: 0 | 1;
+  category: string;
+  charityname: string;
+  city: string;
+  closeschedule: TCloseSchedule[];
+  description: string;
+  kudos: number;
+  lat: number;
+  lng: number;
+  phone: string;
+  schedule: TSchedule[];
+  selectedAll: boolean;
+  service: string;
+  servicetype: string;
+  showflag: boolean;
+  state: string;
+  updateshelter: string;
+  useremail: string;
+  userid: string;
+  website: string;
+  zip: number;
+}
+
+export interface TCloseSchedule {
+  day: TDay;
+  period: string;
+  type: string;
+}
+
+export interface TSchedule {
+  day: TDay;
+  fromstring: string;
+  tostring: string;
+  type: string;
+}
+
+export interface THomeButton {
+  text: string;
+  icon: object;
```

Comment:
Then, in `Icons.tsx`, give a type hint to TS like so:

```
export const CallIcon: React.ReactElement<SvgIconProps> = <Call />;
export const RestaurantIcon: React.ReactElement<SvgIconProps> = <Restaurant />;
```

otherwise, TS just assumes they are generic TSX elements

Reply:
updated 👍